TFNetworkLayer.py

```diff
@@ -2867,14 +2868,24 @@ def __init__(self, axis=None, energy_factor=None,
           tf.ones_like(window_start) * (energy_shape[axis] - window_len),
           window_start)  # case 2.
         n_batch = energy_shape[energy_data.batch_dim_axis]
-        indices = expand_dims_unbroadcast(tf.range(window_len), energy_data.batch_dim_axis, n_batch)
-        # time-major: (W,B) + (1,B), batch-major: (B, W) + (1,B)
-        indices += tf.expand_dims(tf.to_int32(window_start), axis=axis)
-        idxs = nd_indices(indices)
-        mask_shape = energy_shape[:2]  # (T, B)
-        mask_shape[axis] = window_len
-        energy_mask_window = tf.scatter_nd(idxs, tf.ones(shape=mask_shape), energy_shape[:2])
+        indices = expand_dims_unbroadcast(tf.range(window_len), 0, n_batch)  # (B, W)
+        # (B, W) + (B, 1)
+        indices += tf.expand_dims(tf.to_int32(window_start), axis=1)  # + (B, 1)
+        indices.set_shape((None, None))  # required for older TF versions
+        idxs = nd_indices(indices)  # (B, W, 2)
+        mask_shape = [n_batch, window_len]  # (B, W)
+        rem_energy_data = energy_data.copy_with_time_dim_axis(time_dim_axis=1)  # (B, T, ...)
+        rem_energy_shape = tf.shape(rem_energy_data.placeholder, name="rem_energy_shape")
```

Comment:
Note that I prefer to use `TFUtil.get_shape` instead of `tf.shape` in many cases now.

Reply:
Cool, I changed that.